engine/classes/Elgg/EventsService.php

```diff
@@ -44,11 +44,29 @@ public function trigger($event, $type, $object = null, array $options = array())
 		$args = array($event, $type, $object);
 
 		foreach ($events as $callback) {
+
+			$callback_copy = $callback;
+
+			if (is_string($callback)
+					&& false !== strpos($callback, '::')
+					&& !function_exists($callback)
+					&& class_exists($callback)) {
```

Comment:
This isn't very readable. How about moving these checks into a separate method? (```isClassCallback()``` or something)